dbeam-core/src/main/java/com/spotify/dbeam/args/DbeamQueryBuilder.java

```diff
@@ -0,0 +1,184 @@
+    if (uppSql.toUpperCase().contains("WHERE ")) {
+      isContainsWhere = true;
+    }
+
+    fromIdx = uppSql.indexOf("FROM");
+    if (fromIdx < 0) {
+      throw new IllegalArgumentException("Sql query missing FROM clause");
+    }
+    
+    // TODO: may be check that LIMIT is not present;
+    //  can be difficult to judge in a complicated query
+
+    this.sqlBuilder = new StringBuilder(sqlQuery);
+    
+    if (!isContainsWhere) {
+      sqlBuilder.append(" WHERE 1=1");
+    }
+
+  }
+
+  private DbeamQueryBuilder(DbeamQueryBuilder that) {
+    this.sqlBuilder = new StringBuilder(that.sqlBuilder);
+    this.fromIdx = that.fromIdx;
+    this.limitStr = that.limitStr;
+  }
+
+  public DbeamQueryBuilder copy() {
+    return new DbeamQueryBuilder(this);
+  }
+
+  public static DbeamQueryBuilder fromTablename(final String tableName) {
+    return new DbeamQueryBuilder(String.format("SELECT * FROM %s WHERE 1=1", tableName));
+  }
+
+  public static DbeamQueryBuilder fromSqlQuery(final String sqlQuery) {
+    String s = removeTrailingSymbols(sqlQuery);
+    return new DbeamQueryBuilder(s);
+  }
+
+  public DbeamQueryBuilder withPartitionCondition(
+          String partitionColumn, String startPointIncl, String endPointExcl) {
+    sqlBuilder.append(createSqlPartitionCondition(partitionColumn, startPointIncl, endPointExcl));
+    return this;
+  }
+          
+  // TODO It is assumed now that partitionColumn is of non-numeric type 
+  private static String createSqlPartitionCondition(
+      String partitionColumn, String startPointIncl, String endPointExcl) {
+    return String.format(
+        " AND %s >= '%s' AND %s < '%s'",
+        partitionColumn, startPointIncl, partitionColumn, endPointExcl);
+  }
+
+  public DbeamQueryBuilder withParallelizationCondition(
+      String partitionColumn, long startPointIncl, long endPoint, boolean isEndPointExcl) {
+    sqlBuilder.append(
+        createSqlSplitCondition(partitionColumn, startPointIncl, endPoint, isEndPointExcl));
+    return this;
+  }
+
+  private static String createSqlSplitCondition(
+          String partitionColumn, long startPointIncl, long endPoint, boolean isEndPointExcl) {
+    
+    String upperBoundOperation = isEndPointExcl ? "<" : "<=";
+    return String.format(
+            " AND %s >= %s AND %s %s %s",
+            partitionColumn, startPointIncl, partitionColumn, upperBoundOperation, endPoint);
+  }
+
+  /**
+   * Returns generated SQL query string.
+   * 
+   * @return generated SQL query string.
+   */
+  public String build() {
+    limitStr.map(x -> sqlBuilder.append(x));
+    limitStr = Optional.empty();
+    return sqlBuilder.toString();
+  }
+
+  private static String removeTrailingSymbols(String sqlQuery) {
+    return sqlQuery.replaceAll("[\\s|;]+$", "");
+  }
+
+  public DbeamQueryBuilder withLimit(Optional<Integer> limitOpt) {
+    return limitOpt.map(l -> this.withLimit(l)).orElse(this);
+  }
+  
+  public DbeamQueryBuilder withLimit(long limit) {
```

Comment:
Lets keep it as `long` and change 
```
 public DbeamQueryBuilder withLimit(Optional<Integer> limitOpt)
``` 
to 
```
 public DbeamQueryBuilder withLimit(Optional<Long> limitOpt)
```

Reply:
Agree.